xoinvader/background.py

```diff
@@ -0,0 +1,261 @@
+from xoinvader.render import Renderable
+
+
+class Chunk(object):
+    """Class for storing background chunks.
+
+    :param name: name of the chunk
+    :type name: string
+
+    :param lines: list of all chunk lines
+    :type lines: list
+
+    :param length: length of lines list (for convenience)
+    :type length: integer
+
+    :param __MAGIC__: magic of chunk boundary in the file
+    :type __MAGIC__: string
+    """
+
+    __MAGIC__ = "~chunk~"
+
+    def __init__(self, name):
+        self.name = name
+        self.lines = []
+        self.length = 0
+
+    def add_line(self, line):
+        """Add line to `lines` list of the chunk. Increases `length` by 1.
+
+        :param line: line to add
+        :type line: string
+        """
+        self.lines.append(line)
+        self.length += 1
+
+    def __getitem__(self, index):
+        return self.lines[index]
+
+    def __setitem__(self, index, item):
+        self.lines[index] = item
+
+
+def load_chunks(filename, trim_width=None):
+    """Load chunks from file.
+
+    File format:
+    ```
+    -~chunk~- <chunk_name>
+    <data>
+    ...
+    ```
+
+    `~chunk~` is magic string, and chunk spans until next magic line, or
+    until end of file.
+
+    Completely blank lines are ignored.
+    If `trim_width` is provided, all lines are truncated to this length.
+
+    :return: list of loaded chunks
+    :rtype: list
+
+    :param filename: path to file to load chunks from
+    :type filename: string
+
+    :param trim_width: width to truncate long lines to
+    :type trim_width: integer"""
+
+    with open(filename, "r") as f:
+        chunks = []
+        current_chunk = None
+        names = []
+        for line in f:
+            line = line.rstrip('\n')
+            if not line:
+                continue
+            elif line.startswith(Chunk.__MAGIC__):
+                name_split = line.split()
+                if len(name_split) < 2:
+                    raise ValueError("Chunks with empty names are not allowed")
+                name = name_split[1].strip()
+                if name in names:
+                    raise ValueError("Name {} already defined in background "
+                                     "file {}".format(name, filename))
+                current_chunk = Chunk(name)
+                chunks.append(current_chunk)
+                names.append(name)
+                continue
+            elif not current_chunk:
+                continue
+            # ok, now we got both chunk and something to store in it
+            if trim_width:
+                line_add = line[:trim_width]
+            else:
+                line_add = line
+            current_chunk.add_line(line_add)
+        if not chunks:
+            raise ValueError("File {} does not contain "
+                             "any chunks".format(filename))
+        return chunks
+
+
+class Background(Renderable):
+    """Class for storing and managing level background.
```

Comment:
This docstrings like commit messages, some sort of equal rules.
First string is header sentence with terse description ended by dot, then empty line and other somehow project-depend formatted detailes.